consumer/consumererror/scrapeerrors.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import (
+	"fmt"
+)
+
+// ScrapeErrors contains multiple PartialScrapeErrors and can also contain generic errors.
+type ScrapeErrors struct {
+	errs              []error
+	failedScrapeCount int
+}
+
+// Add adds a PartialScrapeError with the provided failed count and error.
+func (s *ScrapeErrors) Add(failed int, err error) {
+	s.errs = append(s.errs, NewPartialScrapeError(err, failed))
+	s.failedScrapeCount += failed
+}
+
+// Addf adds a PartialScrapeError with the provided failed count and arguments to format an error.
+func (s *ScrapeErrors) Addf(failed int, format string, a ...interface{}) {
+	s.errs = append(s.errs, NewPartialScrapeError(fmt.Errorf(format, a...), failed))
```

Comment:
Do we need `PartialScrapeError` to be exposed anymore? We should not remove it in this PR, but want to make sure we have a plan either remove it in a followup, or keep it there and understand why is necessary.

Reply:
We use `Failed` field of `PartialScrapeError` in other packages:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/obsreport/obsreport_scraper.go#L104

So we need to introduce a method to retrieve this value and maybe add an interface instead of the exported  `PartialScrapeError` struct.